### PR TITLE
Migrate free-range components

### PR DESCRIPTION
**Pull request by Etheryte**
## What does this PR change?

~~Blocked by waiting on the merge of https://github.com/uyuni-project/uyuni/pull/3180~~

Migrate all components that aren't part of a subdirectory to Typescript.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Part of: https://github.com/SUSE/spacewalk/issues/13739
Fixes: https://github.com/SUSE/spacewalk/issues/13931

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by Etheryte** [comment]
The previous PR got merged so this is no ready for review.

File: web/html/src/components/ace-editor.tsx [hunk 1]
```diff
@@ -7,25 +6,15 @@ type Props = {
   minLines: number;
   maxLines: number;
   readOnly: boolean;
-  onChange: (...args: any[]) => any;
+  onChange?: (...args: any[]) => any;
   className: string;
   id: string;
   content: React.ReactNode;
+  /** Obsolete, do not use */
+  name?: string;
```
Comment (hustodemon) [inline web/html/src/components/ace-editor.tsx:14]: Looks like the only usage of the ace editor is in the `config-channels.tsx`: we pass the `name`, but we don't use it in the component. Can we remove it?

**Review by hustodemon — APPROVED** [review]
Thanks for the big effort @Etheryte :+1:  Looks good to me, I only left 1 minor comment to check.

Have you run the whole cucumber testsuite on your branch? It'd be worth it as it could reveal any possible unpleasant surprises. Just ping me if you need more info about that.

File: web/html/src/components/ace-editor.tsx (hunk 1 above)
Comment (Etheryte) [reply web/html/src/components/ace-editor.tsx]: Good shout, thanks for pointing this out. Might even be worth keeping an eye out in the future, perhaps we can cut this dependency out altogether at some point.

**Comment by Etheryte** [comment]
I haven't run Cucumber on this branch alone, but almost all of the changes with all TS migrations should be one-to-one, i.e. functionally equivalent.  
I have tested this sporadically manually, but mostly I rely on the TS compiler and JS build to find any issues I might've missed.

**Pull request merged**